RDoc-2883 OpenTelemetry (#1862)
ravendb/docs: Documentation/6.0/Raven.Documentation.Pages/server/administration/monitoring/opentelemetry.markdown

{INFO/}

RavenDB exposes the following metrics:

| Name | Description |
| :--- | :--- |
| ravendb.serverwide.general | Exposes general info about server |
| ravendb.serverwide.requests | Exposes informations about requests processed by server |
| ravendb.serverwide.storage | Exposes storage informations |
| ravendb.serverwide.gc | Exposes detailed informations about Garbage Collector |
| ravendb.serverwide.resources | Exposes detailed information about resources usage (e.g. CPU etc) |
| ravendb.serverwide.totaldatabases | Exposes aggregated informations about databases on the server |
| ravendb.serverwide.cpucredits | Exposes status of CPU credits (cloud) |

We also support exposing metrices developed by Microsoft for AspNetCore and also .NET Runtime.
More info about it can be found on official Microsoft documentations:
- [Runtime documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Runtime#metrics)
- [AspNetCore documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/README.md#metrics)

> ### Ommiting meters

Reviewer: Configuring meters

> Omitting meters is configurable through our standard configuration settings.

Reviewer: By default only most commonly used meters are turned on, but this can be controlled via following configuration options. You can read more about setting the server configuration [here]().

# OpenTelemetry Support
---

{NOTE: }

* OpenTelemetry is a popular monitoring standard designated to help in the inspection and 
  administration of networks, infrastructures, databases, etc.  

* RavenDB sends data metrics via an OpenTelemetry Protocol protocol, 
  allowing a OpenTelemetry retriever to scrape the data from RavenDB.  

* A OpenTelemetry support is provided by RavenDB instances both on-premise and on the cloud.  

* You can also retrieve data for OpenTelemetry collector from Prometheus endpoint.

{NOTE/}

---

{PANEL: OpenTelemetry}

OpenTelemetry is a collection of APIs, SDKs, and tools. Use it to instrument, generate, collect, and export telemetry data (metrics, logs, and traces) to help you analyze your software's performance and behavior. (description via [https://opentelemetry.io](https://opentelemetry.io))

> RavenDB utilize official SDK and allows user to retrieve the metrics via OpenTelemetry protocol and much more!

Reviewer 2: ```suggestion
RavenDB utilizes official SDK and allows user to retrieve the metrics via OpenTelemetry protocol and much more!
```

Reviewer 3: I will be reviewing this article and fixing all phrasing/grammar issues in:
https://issues.hibernatingrhinos.com/issue/RDoc-2944/Review-Open-Telemetry-documentation

| Name                              | Description |
|:----------------------------------| :--- |
| ravendb.server.general            | Exposes general info about server |
| ravendb.server.requests       | Exposes informations about requests processed by server |
| ravendb.server.storage        | Exposes storage informations |
| ravendb.server.gc             | Exposes detailed informations about Garbage Collector |
| ravendb.server.resources      | Exposes detailed information about resources usage (e.g. CPU etc) |
| ravendb.server.totaldatabases | Exposes aggregated informations about databases on the server |
| ravendb.server.cpucredits     | Exposes status of CPU credits (cloud) |

We also support exposing metrices developed by Microsoft for AspNetCore and also .NET Runtime.
More info about it can be found on official Microsoft documentations:
- [Runtime documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Runtime#metrics)
- [AspNetCore documentation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/README.md#metrics)

### Configuring meters

> By default, only most commonly used meters are turned on, but this can be controlled via following configuration options: 

Reviewer 2: ```suggestion
By default, only most commonly used meters are enabled. This can be controlled via following configuration options: 
```

| Configuration name | Meter name | Default value |
| :--- | :--- | :--- |
| Monitoring.OpenTelemetry.Meters.AspNetCore.Enabled | Official AspNetCore instrumentation | false |
| Monitoring.OpenTelemetry.Meters.Runtime.Enabled | Official Runtime instrumentation | false |
| Monitoring.OpenTelemetry.Meters.Server.Storage.Enabled | ravendb.server.storage | true |
| Monitoring.OpenTelemetry.Meters.Server.CPUCredits.Enabled | ravendb.server.cpucredits | false|
| Monitoring.OpenTelemetry.Meters.Server.Resources.Enabled | ravendb.server.resources | true |
| Monitoring.OpenTelemetry.Meters.Server.TotalDatabases.Enabled | ravendb.server.totaldatabases | true |
| Monitoring.OpenTelemetry.Meters.Server.Requests.Enabled | ravendb.server.requests | true |
| Monitoring.OpenTelemetry.Meters.Server.GC.Enabled | ravendb.server.gc | false |

> ### Meters instruments

Reviewer 2: are we exposing information about the client certificates expiration?

| Name | Description | Instrument type |
| :--- | :--- | :--- |
| ravendb.server.cpucredits.alert_raised | CPU Credits Any Alert Raised | Gauge |
| ravendb.server.cpucredits.background.tasks.alert_raised | CPU Credits Background Tasks Alert Raised | Gauge |
| ravendb.server.cpucredits.base | CPU Credits Base | UpDownCounter |
| ravendb.server.cpucredits.consumption_current | CPU Credits Gained Per Second | UpDownCounter |
| ravendb.server.cpucredits.failover.alert_raised | CPU Credits Failover Alert Raised | Gauge |
| ravendb.server.cpucredits.max | CPU Credits Max | UpDownCounter |
| ravendb.server.cpucredits.remaining | CPU Credits Remaining | Gauge |
| ravendb.server.gc.compacted | Specifies if this is a compacting GC or not. | Gauge |
| ravendb.server.gc.concurrent | Specifies if this is a concurrent GC or not. | Gauge |
| ravendb.server.gc.finalizationpendingcount | Gets the number of objects ready for finalization this GC observed. | Gauge |

> | ravendb.server.gc.fragmented | Gets the total fragmentation (in MB) when the last garbage collection occurred. | Gauge |

Reviewer 4: does open telemetry suggests to use explicit units just like prometheus metrics? 
See: https://prometheus.io/docs/practices/naming/

Author: I don't see it here: https://opentelemetry.io/docs/specs/semconv/general/metrics/

| ravendb.server.requests.total.requests | Total number of requests since server startup | UpDownCounter |
| ravendb.server.storage.storage.disk.ios.read_operations | IO read operations per second | Gauge |
| ravendb.server.storage.storage.disk.ios.write_operations | IO write operations per second | Gauge |
| ravendb.server.storage.storage.disk.queue_length | Queue length | Gauge |
| ravendb.server.storage.storage.disk.read_throughput | Read throughput in kilobytes per second | Gauge |
| ravendb.server.storage.storage.disk.remaining.space | Remaining server storage disk space in MB | Gauge |
| ravendb.server.storage.storage.disk.remaining.space_percentage | Remaining server storage disk space in % | Gauge |
| ravendb.server.storage.storage.disk.write_throughput | Write throughput in kilobytes per second | Gauge |
| ravendb.server.storage.storage.total_size | Server storage total size in MB | Gauge |
| ravendb.server.storage.storage.used_size | Server storage used size in MB | Gauge |
| ravendb.server.totaldatabases.count_stale_indexes | Number of stale indexes in all loaded databases | UpDownCounter |

> | ravendb.server.totaldatabases.data.written.per_second | Number of bytes written \(documents, attachments, counters\) in all loaded databases | Gauge |

Reviewer 4: does it include time series writes?

| ravendb.server.totaldatabases.database.loaded_count | Number of loaded databases | UpDownCounter |
| ravendb.server.totaldatabases.database.node_count | Number of databases for current node | UpDownCounter |
| ravendb.server.totaldatabases.database.total_count | Number of all databases | UpDownCounter |
| ravendb.server.totaldatabases.map_reduce.index.mapped_per_second | Number of maps per second for map-reduce indexes \(one minute rate\) in all loaded databases | Gauge |
| ravendb.server.totaldatabases.map_reduce.index.reduced_per_second | Number of reduces per second for map-reduce indexes \(one minute rate\) in all loaded databases | Gauge |
| ravendb.server.totaldatabases.map.index.indexed_per_second | Number of indexed documents per second for map indexes \(one minute rate\) in all loaded databases | Gauge |
| ravendb.server.totaldatabases.number_error_indexes | Number of error indexes in all loaded databases | UpDownCounter |
| ravendb.server.totaldatabases.number_of_indexes | Number of indexes in all loaded databases | UpDownCounter |
| ravendb.server.totaldatabases.number.faulty_indexes | Number of faulty indexes in all loaded databases | UpDownCounter |

> | ravendb.server.totaldatabases.writes_per_second | Number of writes \(documents, attachments, counters\) in all loaded databases | Gauge |

Reviewer 4: does it include time series writes?

Author: This is from Snmp description, and it contains. Fix: https://github.com/ravendb/ravendb/pull/19141

Reviewer 5: So it should be reflected in the docs as well, right?

Reviewer 3: I have added that in my latest PRs:
https://github.com/ravendb/docs/pull/1894/files
&
https://github.com/ravendb/docs/pull/1893/files

Reviewer 5: Great. Thanks
